SDKBuildScripts/unity_buildAppCenterTestIOS.sh

```diff
@@ -24,94 +24,116 @@ ProjectFolderName=$(basename "$XCodeWorkspaceDirectory")
+        git push -u origin "$AppCenterGitRepoBranchName"
+        echo 'ERROR: Unity Job '"$AppCenterGitRepoBranchName"' did not yet exist.'
+        echo "The branch has been created and populated, but must be manually configured in AppCenter."
+        echo "Please try again once the AppCenter build branch has been properly configured."
+        exit 1
+    fi
+
+    git push
+}
 
 #queue the appcenter build
-appcenter build queue --app "PlayFabSDKTeam/PlayFabUnityXCode" --branch $AppCenterGitRepoBranchName --quiet -d 
-if [ $? -ne 0 ]; then
-echo "Error queueing build!"
-false
-exit
-fi
+QueueAppCenterBuild() {
+    appcenter build queue --app "PlayFabSDKTeam/PlayFabUnityXCode" --branch $AppCenterGitRepoBranchName --quiet -d 
+    if [ $? -ne 0 ]; then
+        echo "Error queueing build!"
+        exit 1
+    fi
 
-BuildStatusJSON=$(appcenter build branches show -b $AppCenterGitRepoBranchName -a "PlayFabSDKTeam/PlayFabUnityXCode" --quiet --output json)
-BuildStatus="\"notStarted\""
+    BuildStatusJSON=$(appcenter build branches show -b $AppCenterGitRepoBranchName -a "PlayFabSDKTeam/PlayFabUnityXCode" --quiet --output json)
+    BuildStatus="\"notStarted\""
+}
 
 #monitor the status of the build and wait until it is complete.
-while [ "$BuildStatus" != "\"completed\"" ]
-do 
-sleep 10
-BuildStatusJSON=$(appcenter build branches show -b $AppCenterGitRepoBranchName -a "PlayFabSDKTeam/PlayFabUnityXCode" --quiet --output json)
-BuildStatus=$(echo "$BuildStatusJSON" | jq .status)
-echo "Build Status: $BuildStatus"
-done
-
-#extract the results and build number
-BuildResult=$(echo "$BuildStatusJSON" | jq .result)
-BuildNumber=$(echo "$BuildStatusJSON" | jq .buildNumber | sed s/\"//g)
-
-echo "Build $BuildNumber has $BuildResult."
+WaitForAppCenterBuild() {
+    for i in {1..30}
+    do
+        sleep 1
+        BuildStatusJSON=$(appcenter build branches show -b $AppCenterGitRepoBranchName -a "PlayFabSDKTeam/PlayFabUnityXCode" --quiet --output json)
+        BuildStatus=$(echo "$BuildStatusJSON" | jq .status)
+        echo "WaitForAppCenterBuild check number: $i, Build Status: $BuildStatus"
+        if [ "$BuildStatus" = "\"completed\"" ]; then
+            return 0
+        fi
+    done
+    
+    exit 1
+}
+
+ExtractBuildResults() {
+    #extract the results and build number
+    BuildResult=$(echo "$BuildStatusJSON" | jq .result)
+    BuildNumber=$(echo "$BuildStatusJSON" | jq .buildNumber | sed s/\"//g)
+
+    echo "Build $BuildNumber has $BuildResult."
+}
 
 #Download the build if successful, or print the logs if not.
-if [ "$BuildResult" = "\"succeeded\"" ]; then
-#Return the appcenter build repo to a clean state for next time.
-git reset --hard $AppCenterGitRepoCleanTag
-git push --force 
-popd
-appcenter build download --type build --app "PlayFabSDKTeam/PlayFabUnityXCode" --id $BuildNumber --file PlayFabIOS.ipa
-else
-popd
-appcenter build logs --app "PlayFabSDKTeam/PlayFabUnityXCode" --id $BuildNumber
-false
-fi
+FinalCleanup() {
```

Comment:
FinalCleanup is a little too vague, imo.  The most important thing this function does is downloads the IPA if it passes.